WcaOnRails/app/views/shared/_associated_events_picker.html.erb

```diff
@@ -0,0 +1,36 @@
+<%#
+Variables:
+ - form_builder
+ - label
+ - disabled
+ - events_association_name -> name of the associated events relation
+ - allowed_events -> an Array of Event objects that should be shown (all by default)
+ - selected_events -> an Array of Event objects that whould be marked as selected
+%>
+
+<% label ||= "Events" %>
+<% disabled ||= false %>
+<% allowed_events ||= Event.all_official %>
+
+<%# Note: form_builder.object is the object having associated events. %>
+<% associated_events = form_builder.object.send(events_association_name)
+   all_events = allowed_events.map do |event|
+     associated_events.find_by_event_id(event.id) || associated_events.build(event_id: event.id)
+   end %>
+
+<div class="form-group">
+  <%= label_tag "associated_events", label %>
+  <div id="associated_events">
```

Comment:
Maybe the id should be the `events_association_name`. So we would have `<div id="registration_events">` in this case. I guess that there won't be multiple inputs for the same relation.